src/generator/AutoRest.CSharp/ClientModelExtensions.cs

```diff
@@ -403,60 +415,49 @@ public static string CheckNull(string valueReference, string executionBlock)
         }
 
 
-        private static void AppendConstraintValidations(string valueReference, Dictionary<Constraint, string> constraints, IndentedStringBuilder sb)
+        private static void AppendConstraintValidations(string valueReference, Dictionary<Constraint, string> constraints, IndentedStringBuilder sb, KnownFormat format)
         {
             foreach (var constraint in constraints.Keys)
             {
                 string constraintCheck;
-                string constraintValue = constraints[constraint];
+                string constraintValue = (format == KnownFormat.@char) ?$"'{constraints[constraint]}'" : constraints[constraint];
                 switch (constraint)
                 {
                     case Constraint.ExclusiveMaximum:
-                        constraintCheck = string.Format(CultureInfo.InvariantCulture, "{0} >= {1}", valueReference,
-                            constraints[constraint]);
+                        constraintCheck = $"{valueReference} >= {constraintValue}";
```

Comment:
Yeah, me too. It's far less complicated to see, and most of the time it runs faster than format anyway. :D